internal/commands/vaultsecrets/secrets/delete.go

```diff
@@ -64,6 +65,7 @@ func NewCmdDelete(ctx *cmd.Context, runF func(*DeleteOpts) error) *cmd.Command {
 			return deleteRun(opts)
 		},
 	}
+	cmd.Args.Autocomplete = helper.PredictSecretName(ctx, cmd, opts.PreviewClient)
```

Comment:
Thanks @dadgar. What changes when defined like this vs inline before?

Reply:
I needed access to the `cmd` variable, so it couldn't be defined in line.

Reply:
I see the cmd injection logic, but I didn't need it as such when using the profile set app name. I am trying to understand how appname is populated now vs if when we were inlining autocomplete.

Reply:
Gotcha. I think my original question is more on the lines that `AutoComplete` is always going to need these flag or ability to parse these flags. If we need to set this up this way for all commands, then inlining use case is really not there. Maybe we have need a setter construct and autocomplete becomes internal detail. This gets job done, but the whole app name lookup under persistence flag also needs to hide internally in provided frameworks (on the other comment). For now, this is ok.